core/common/src/main/java/alluxio/collections/IndexedSet.java

```
@@ -255,46 +231,40 @@ public void remove() {
   }
 
   /**
-   * Whether there is an object with the specified field value in the set.
+   * Whether there is an object with the specified unique index field value in the set.
    *
-   * @param index the field index
+   * @param indexDefinition the field index definition
    * @param value the field value
    * @return true if there is one such object, otherwise false
    */
-  public boolean contains(FieldIndex<T> index, Object value) {
-    ConcurrentHashSet<T> set = getByFieldInternal(index, value);
-    return set != null && !set.isEmpty();
+  public boolean contains(IndexDefinition<T> indexDefinition, Object value) {
+    FieldIndex<T> index = mIndices.get(indexDefinition);
+    return index != null && index.contains(value);
```

we should probably throw an exception if the index can't be found.